src/relay.js

```diff
@@ -20,41 +20,59 @@ import _ from 'lodash';
 import simplifyAST from './simplifyAST';
 
 class NodeTypeMapper {
-  constructor(sequelize) {
-    this.models = Object.keys(sequelize.models);
-    this.models.forEach(model => {
-      this[model] = null;
-    });
+  constructor() {
+    this.map = { };
   }
 
   mapTypes(types) {
-    Object.keys(types).forEach(type => {
-      this[type] = types[type];
+    Object.entries(types).forEach(([k, v]) => {
+      this.map[k] = v.type
+        ? v
+        : { type: v };
     });
   }
+
+  item(type) {
+    return this.map[type];
+  }
 }
 
 export function idFetcher(sequelize, nodeTypeMapper) {
   return globalId => {
-    let {type, id} = fromGlobalId(globalId);
-    const models = Object.keys(sequelize.models);
-    if (models.some(model => model === type)) {
-      return sequelize.models[type].findById(id);
-    }
-    if (nodeTypeMapper[type]) {
-      return nodeTypeMapper[type];
+    const {type, id} = fromGlobalId(globalId);
+
+    const nodeType = nodeTypeMapper.item(type);
+    if (nodeType && typeof nodeType.resolve === 'function') {
+      const res = nodeType.resolve(globalId);
+      res.__graphqlType__ = type;
+      return res;
```

Comment:
We need to add a test that uses an async resolver since i'm not sure this code will work for that. (We need to wait for the promise before adding type)